src/Npgsql/PostgresException.cs

```diff
@@ -271,6 +271,42 @@ public override bool IsTransient
         /// </summary>
         public NpgsqlStatement Statement { get; internal set; }
 
+        /// <inheritdoc />
+        public override string ToString()
+        {
+            var builder = new StringBuilder(base.ToString());
+            builder.AppendLine().Append("  Exception data:");
+
+            void AppendPropertyLine<T>(string propertyName, T propertyValue)
+            {
+                if (propertyValue == default)
+                {
+                    return;
+                }
+
+                builder.AppendLine().Append("    ").Append(propertyName).Append(": ").Append(propertyValue);
```

Comment:
But when you use generics, boxing appears here. I don't think that's an issue.

Use `AppendLine("    ")`. It does the same thing as calling `AppendLine` and `Append`, but it's shorter.

Reply:
I am creating the new line first and then writing the whitespace after. `AppendLine("      ")` would do the opposite.

Reply:
Ah, yes. My mistake.